implementation.d.ts

```diff
@@ -0,0 +1,54 @@
+interface PromiseResolution<T> {
+	status: 'fulfilled';
+	value: T;
+}
+
+interface PromiseRejection<E extends Error = Error> {
```

Comment:
Promises need not reject with an error; this should just be E.

Reply:
Technically true, but a bad practice since you will get no information on where the error originated from due to lack of a stack trace.

I would argue going the stricter way, since it is common with TypeScript. While the following code works in JS:

```javascript
parseInt(10, 10)
```

You would get an error in TypeScript:

```
error TS2345: Argument of type '10' is not assignable to parameter of type 'string'.
parseInt(10, 10)
         ~~
```

On the other hand, TypeScript does allow you to pass non-errors to the `throw` operator.

I'll leave the final decision on you.

Reply:
Whatever throw allows is what a promise rejection allows; it’s a type bug if you pass a number to parseInt but not if you reject with `false`, say.

Reply:
Very well.